spring-boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaAutoConfiguration.java

```diff
@@ -0,0 +1,214 @@
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.json.MetricsModule;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.common.base.Strings;
+
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponse;
+import com.linecorp.armeria.common.http.HttpResponseWriter;
+import com.linecorp.armeria.common.http.HttpSessionProtocols;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.docs.DocServiceBuilder;
+import com.linecorp.armeria.server.http.AbstractHttpService;
+import com.linecorp.armeria.server.http.healthcheck.HealthChecker;
+import com.linecorp.armeria.server.http.healthcheck.HttpHealthCheckService;
+import com.linecorp.armeria.server.logging.DropwizardMetricCollectingService;
+import com.linecorp.armeria.spring.ArmeriaSettings.Port;
+
+import io.netty.util.NetUtil;
+
+@Configuration
+@EnableConfigurationProperties(ArmeriaSettings.class)
+@ConditionalOnMissingBean(Server.class)
+public class ArmeriaAutoConfiguration {
+
+    private static final Logger logger = LoggerFactory.getLogger(ArmeriaAutoConfiguration.class);
+
+    private static final HealthChecker[] EMPTY_HEALTH_CHECKERS = new HealthChecker[0];
+
+    private static final Port DEFAULT_PORT = new Port().setPort(8080)
+                                                       .setProtocol(HttpSessionProtocols.HTTP);
+
+    @Bean
+    @ConditionalOnMissingBean(MetricRegistry.class)
+    MetricRegistry metricRegistry() {
+        return new MetricRegistry();
+    }
+
+    /**
+     * Create a {@link Server} bean.
+     */
+    @Bean
+    public Server armeriaServer(
+            ArmeriaSettings armeriaSettings,
+            MetricRegistry metricRegistry,
+            Optional<List<HealthChecker>> healthCheckers,
+            Optional<List<ArmeriaServerConfigurator>> armeriaServiceInitializers,
+            Optional<List<ThriftServiceRegistrationBean>> thriftServiceRegistrationBeans,
+            Optional<List<HttpServiceRegistrationBean>> httpServiceRegistrationBeans)
+            throws InterruptedException {
+        if (!armeriaServiceInitializers.isPresent() &&
+            !thriftServiceRegistrationBeans.isPresent() &&
+            !httpServiceRegistrationBeans.isPresent()) {
+            // No services to register, no need to start up armeria server.
+            return null;
+        }
+
+        ServerBuilder server = new ServerBuilder()
+                .gracefulShutdownTimeout(
+                        armeriaSettings.getGracefulShutdownQuietPeriodMillis(),
```

Comment:
As you know, I personally love forcing good opinions on users :) But I guess spring usually does provide options to turn off defaults. How about, e.g. echecking for -1 graceful shutdown, adding a boolean flag for dropwizard decorators, and checking for null for health check path to allow disabling them. Having the defaults enable all these features sounds great to me though.

Of course, if armeria declared itself as an opinionated framework (e.g., https://github.com/curioswitch/curiostack/tree/master/common/server/framework `</plug>`, this isn't necessary, but I think it's general direction is flexibility, not opinionatedness.